pkg/flow/module.go

```diff
@@ -171,4 +173,8 @@ type moduleControllerOptions struct {
 	// ServiceMap is a map of services which can be used in the module
 	// controller.
 	ServiceMap controller.ServiceMap
+
+	// WorkerPool is a worker pool that can be used to run tasks asynchronously. A default pool will be created if this
+	// is nil.
+	WorkerPool worker.Pool
```

Comment:
(premature question, still haven't read through the entire thing)
Is this docstring valid here as well? Or is a nil worker pool replaced only in the root controller?

Reply:
AFAICT, both modules' controllers and the root controller are created via `newController` in `flow.go` which is where the `nil` check is performed. Open to suggestions if there's a problem with this :)